ui/ducks/metamask/metamask.js

```diff
@@ -198,6 +210,49 @@ export default function reduceMetamask(state = {}, action) {
   }
 }
 
+const toHexWei = (value, expectHexWei) => {
+  return addHexPrefix(expectHexWei ? value : decGWEIToHexWEI(value));
```

Comment:
until we get rid of the other code, the component fields are the ones who decide how to convert values, whereas in the new branch we'll just interpret everything in decGWEI until we submit to the network.